### DISCUSSION TITLE
Testing

### DISCUSSION THREAD

**Discussion by robbievanleeuwen** [robbievanleeuwen/section-properties#50] [Ideas]
This discussion is for everything testing related :rocket:

See initial discussion in issue #46.

**Comment by Spectre5** [comment]
@Czarified, just curious about any updates you might have with testing --- issues or progress.

**Reply by Czarified** [reply]
Not much progress, unfortunately. I've basically got hypothesis working, and published the branch @ https://github.com/Czarified/section-properties/branches.

My next work is going to be a specific test on some examples from "Aircraft Structures" by David Peery. It's a common reference within the aerospace industry, and I want to grab some unsymmetric beam examples. I'll write up some doc pages for it too. I've had the book sitting here all week and just haven't actually done it... Eventually! 😄

I'd really like to have some Sphinx pages that get updated with test results whenever we publish a release (or whatever). That seems pretty complicated/impossible, though... Thoughts here?

**Reply by Spectre5** [reply]
Peery is a great text to use for some test cases!

As for the Sphinx bit, I think that is a great idea and something that I also wanted to pursue.  I know it can be done, although I've never tried doing it myself.  I know that [pyvista](https://docs.pyvista.org/examples/index.html) does this - here are their [examples](https://docs.pyvista.org/examples/index.html).  So you might investigate how they do it.  I know other libraries have similar auto-updated examples too!

**Reply by Czarified** [reply]
So... it feels silly to ask this, but I never use sectionproperties to actually compute beam stresses. I just use it for the... section properties... How do I get certain stress results at a certain point on the cross-section? The `get_results()` method seems to just give a list. I guess that's the result at each node? How do I find node numbers?

**Reply by robbievanleeuwen** [reply]
Hi @Czarified, the method you are after is `get_stress()` - docs [here](https://sectionproperties.readthedocs.io/en/latest/rst/post.html#retrieving-cross-section-stress).

Below is an answer I gave to someone via email which I think had the same question you have:

> The get_stress() method will return a list of dictionaries, one for each material in your cross section containing various stresses. Each dictionary entry will give you a list of stresses, e.g. 'sig_vm' will give you a list of the von Mises stresses for the current material. The length of this list will equal the number of nodes in your mesh of the cross section. The ordering of this list is the same as is stored in the CrossSection class, so for example you can look up the mesh_nodes variable in the CrossSection class and this will return the coordinates of the nodes. Note that a zero value in the stress dictionary will be returned if the node does not lie within the specific material. If you only have one material in your cross section you don't have to worry about materials and only one dictionary will be returned from the get_stress() method.

>I hope this makes some sense, in summary get_stress() is what you'll need to use and you can relate these values to the point locations saved in mesh_nodes.

Let me know if you have any further questions 😄

**Reply by Czarified** [reply]
Ahh, thanks Robbie! We may want to add something like this into the library. For now, i've got it in my testing module.

```python
def get_node(nodes, coord) -> (int, tuple):
    '''
    This function will loop over the node list provided,
    finding the index of the coordinates you want.

    Returns the index in the nodes list, and the coords.
    '''
    for index,var in enumerate(nodes):
        if all(var == coord):
            return index, var
        else:
            continue
```

**Reply by Czarified** [reply]
@Spectre5 , Okay, so I've got a test script working and matching Peery over on my branch. I'll work on writing up a docs page this week, and submit a PR for you (and others) to take a look at. It will include some basic hypothesis stuff, too.

**Reply by robbievanleeuwen** [reply]
@Czarified - seems like a sensible addition!
